pkg/utils/json.go

```diff
@@ -42,3 +46,41 @@ func removeNaN(hosts []result.Host) []result.Host {
 	}
 	return newHosts
 }
+
+func GetMetics(searchDir string, m *[]metrics.Metrics) error {
+	resultFilePath := path.Join(path.Dir(path.Clean(searchDir)), "result.txt")
+
+	bytes, err := ioutil.ReadFile(resultFilePath)
+	if err != nil {
+		return err
+	}
+
+	// any line start with '{' and and with '}'
+	r := regexp.MustCompile(`(?m:^{.*}$)`)
+
+	var bm metrics.BaseMetrics
+	for _, jsonBytes := range r.FindAll(bytes, -1) {
+		err := json.Unmarshal(jsonBytes, &bm)
+		if err != nil {
+			fmt.Printf("cannot unmarshal the line '%v' for BaseMetrics: %v\n", string(jsonBytes), err)
+		}
+
+		switch bm.Type {
+		case "metrics.TestDuration":
+			var td metrics.TestDuration
+			err := json.Unmarshal(jsonBytes, &td)
```

Comment:
Does this still unmarshal correctly even though it's a nested struct in the json?

Reply:
Done.
Sorry that it took me this long to understand your comment.